src/vector/layer.rs

```diff
@@ -101,6 +146,13 @@ impl<'a> Layer<'a> {
         _string(rv)
     }
 
+    pub fn test_capability(&self, capability: LayerCaps::Type) -> Result<bool> {
```

Comment:
A couple of minor comments:
- the usual rust convention is to name it `has_capability`
- the argument is of type `&'static str` and hence prevents for eg. testing a string input by the user at runtime.  Would `&'a str` make sense for the method?

Reply:
> the usual rust convention is to name it has_capability

Changed to `has_capability`.

Reply:
> the argument is of type &'static str and hence prevents for eg. testing a string input by the user at runtime. Would &'a str make sense for the method?

I don't know: if it was a pure Rust implementation I would have used an `Enum LayerCaps` and thus enforcing the use of a dedicated type as parameter (and  I'm not sure even why this is a string in gdal ).  

Formally, that does not prevent testing from string input but require a little more work: you need to write a proper parser for the string depending the input come from and/or eventually deal with typo error - which is never obvious to debug.  

I'm  in favor of narrowing the possibilities for safety but I don't mind to change to `&str` if you think it may be useful in other use case.

Reply:
I think it's okay to leave as `&'static str` unless a need to check for dynamic / user-input arbitarry strings arises.